File: neoforge/src/main/java/net/shiroha233/roadweaver/persistence/neoforge/WorldDataHelper.java
package net.shiroha233.roadweaver.persistence.neoforge;

import net.minecraft.server.level.ServerLevel;
import net.minecraft.world.level.Level;
import net.shiroha233.roadweaver.core.model.StructureConnection;
import net.shiroha233.roadweaver.core.model.StructureLocationData;
import net.shiroha233.roadweaver.persistence.WorldDataProvider;

import java.util.ArrayList;
import java.util.List;

/**
 * NeoForge-side convenience accessor for world data.
 */
public final class WorldDataHelper {
    private WorldDataHelper() {}

    public static StructureLocationData getStructureLocations(Level level) {
        if (level instanceof ServerLevel server) {
            return WorldDataProvider.getInstance().getStructureLocations(server);
        }
        return new StructureLocationData(new ArrayList<>());
    }

    public static List<StructureConnection> getConnectedStructures(Level level) {
        if (level instanceof ServerLevel server) {
            return WorldDataProvider.getInstance().getStructureConnections(server);
        }
        return new ArrayList<>();
    }

    public static void setStructureLocations(Level level, StructureLocationData data) {
        if (level instanceof ServerLevel server) {
            WorldDataProvider.getInstance().setStructureLocations(server, data);
        }
    }

    public static void setStructureConnections(Level level, List<StructureConnection> connections) {
        if (level instanceof ServerLevel server) {
            WorldDataProvider.getInstance().setStructureConnections(server, connections);
        }
    }
}

File: neoforge/src/main/kotlin/net/shiroha233/roadweaver/persistence/neoforge/WorldDataHelper.kt
package net.shiroha233.roadweaver.persistence.neoforge

import net.minecraft.server.level.ServerLevel
import net.minecraft.world.level.Level
import net.shiroha233.roadweaver.helpers.Records
import net.shiroha233.roadweaver.persistence.WorldDataProvider

/**
 * NeoForge 端便捷访问工具。
 * 注意：客户端(CLIENT)世界为 ClientLevel，不可直接持久化访问，若传入非 ServerLevel 将返回空数据。
 */
object WorldDataHelper {
    @JvmStatic
    fun getStructureLocations(level: Level): Records.StructureLocationData {
        return if (level is ServerLevel) {
            WorldDataProvider.getInstance().getStructureLocations(level)
        } else {
            Records.StructureLocationData(ArrayList())
        }
    }

    @JvmStatic
    fun getConnectedStructures(level: Level): List<Records.StructureConnection> {
        return if (level is ServerLevel) {
            WorldDataProvider.getInstance().getStructureConnections(level)
        } else {
            ArrayList()
        }
    }

    @JvmStatic
    fun setStructureLocations(level: Level, data: Records.StructureLocationData) {
        if (level is ServerLevel) {
            WorldDataProvider.getInstance().setStructureLocations(level, data)
        }
    }

    @JvmStatic
    fun setStructureConnections(level: Level, connections: List<Records.StructureConnection>) {
        if (level is ServerLevel) {
            WorldDataProvider.getInstance().setStructureConnections(level, connections)
        }
    }
}
